Assets/Mirror/Editor/Weaver/CompilationFinishedHook.cs

```diff
@@ -0,0 +1,147 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using UnityEditor;
+using UnityEditor.Compilation;
+using UnityEngine;
+using Assembly = System.Reflection.Assembly;
+
+namespace Mirror.Weaver
+{
+    internal class ComplilationFinishedHook
+    {
+        [InitializeOnLoadMethod]
+        static void OnInitializeOnLoad()
+        {
+            CompilationPipeline.assemblyCompilationFinished += OnCompilationFinished;
```

Comment:
and this works with first load?

Also,  are you adding the MIRROR define?

Reply:
adding no defines. this works with first load yes.

Reply:
can you please make sure the define is added back?   the motivation for the define was not the weaver,  but to help asset developers port to Mirror.

Reply:
we don't add the define in weaver at the moment. that was only in your sourced based weaver pr.
preprocessor has initializeonload: https://github.com/vis2k/Mirror/blob/master/Assets/Mirror/Editor/PreprocessorDefine.cs